src/logic/safe/store/reducer/safe.ts

```diff
@@ -84,11 +84,16 @@ export default handleActions<AppReduxState['safes'], Payloads>(
       const safeAddress = safe.address
 
       const shouldUpdate = shouldSafeStoreBeUpdated(safe, state.getIn(['safes', safeAddress]))
+      let loadedViaUrl = safe.loadedViaUrl
+
+      if (!state.hasIn(['safes', safeAddress])) {
+        loadedViaUrl = !safe?.name || safe?.name === 'LOADED SAFE'
+      }
 
       return shouldUpdate
         ? state.updateIn(
             ['safes', safeAddress],
-            makeSafe({ name: safe?.name || 'LOADED SAFE', address: safeAddress }),
+            makeSafe({ name: safe?.name || 'LOADED SAFE', address: safeAddress, loadedViaUrl }),
```

Comment:
just a tiny change, can we move the literal to a constant? and use it everywhere?